vibe_code_apps_19/tests/test_session_workspace.py

```
@@ -0,0 +1,186 @@
+    assert not path_is_inside(dest, dest.parent)
+
+
+def test_extract_cannot_escape_session_workspace(tmp_path: Path) -> None:
+    from app.package_io import PackageError, extract_package_zip
+
+    dest = package_dir("11" * 16, temp_dir=tmp_path)
+    buf = io.BytesIO()
+    with zipfile.ZipFile(buf, "w") as zf:
+        zf.writestr("../evil.csv", "a,b\n1,2\n")
+        zf.writestr("manifest.json", "{}")
+    with pytest.raises(PackageError, match="traversal|Absolute|rejected|inside"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)